### PR TITLE
RELATED: RAIL-3794 - Mark up essential APIs as public

### PR DESCRIPTION
**Pull request by lupko**
---

Supported PR commands:

| Command                  | Description            |
| ------------------------ | ---------------------- |
| `ok to test`             | Re-run standard checks |
| `extended test`          | BackstopJS tests       |
| `extended check sonar`   | SonarQube tests        |
| `extended check cypress` | Cypress E2E tests      |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5920/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7701/ : FAILURE in 4m 55s

File: libs/sdk-ui-loaders/api/sdk-ui-loaders.api.md [hunk 1]
```diff
@@ -78,33 +78,33 @@ export interface IDashboardLoader {
...
+// @public
 export interface IDashboardLoadOptions extends IDashboardBasePropsForLoader {
     adaptiveLoadOptions?: AdaptiveLoadOptions;
     clientWorkspace?: IClientWorkspaceIdentifiers;
     extraPlugins?: IEmbeddedPlugin | IEmbeddedPlugin[];
     loadingMode?: "adaptive" | "staticOnly";
 }
 
-// @alpha (undocumented)
+// @public (undocumented)
```
Comment (no23reason) [inline libs/sdk-ui-loaders/api/sdk-ui-loaders.api.md:89]: Public thing without docs, please check it.

File: libs/sdk-ui-loaders/api/sdk-ui-loaders.api.md [hunk 2]
```diff
@@ -55,18 +55,18 @@ export type DashboardLoadResult = {
...
 export const DashboardStub: React_2.FC<IDashboardStubProps>;
 
-// @alpha
+// @public
 export interface IDashboardBasePropsForLoader extends Omit<IDashboardBaseProps, "dashboard"> {
     dashboard: ObjRef;
 }
 
-// @alpha (undocumented)
+// @public (undocumented)
```
Comment (no23reason) [inline libs/sdk-ui-loaders/api/sdk-ui-loaders.api.md:69]: Public thing without docs, please check it.

File: libs/sdk-ui-loaders/api/sdk-ui-loaders.api.md (hunk 1 above)
Comment (no23reason) [reply libs/sdk-ui-loaders/api/sdk-ui-loaders.api.md:89]: There are actually several reported in this file, can you please check them?

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5929/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7710/ : SUCCESS in 16m 24s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: libs/sdk-ui-loaders/api/sdk-ui-loaders.api.md (hunk 1 above)
Comment (lupko) [reply libs/sdk-ui-loaders/api/sdk-ui-loaders.api.md:87]: fixed

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5942/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7723/ : SUCCESS in 16m 05s

File: libs/sdk-ui-loaders/src/dashboard/dashboardLoader.tsx [hunk 3]
```diff
@@ -95,7 +95,13 @@ const AdaptiveLoadStrategies = (
 };
 
 /**
- * @alpha
+ * Default implementation of the {@link IDashboardLoader} interface. This class implements all the
+ * necessary functionality related to either static or dynamic dashboard loading.
+ *
+ * It is possible to construct loader that will be constrained to only static loading of dashboards,
+ * meaning
```
Comment (no23reason) [inline libs/sdk-ui-loaders/src/dashboard/dashboardLoader.tsx:102]: Missing the rest of the sentence.

Comment (lupko) [reply libs/sdk-ui-loaders/src/dashboard/dashboardLoader.tsx]: fixed

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5945/ : SUCCESS in 14s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7725/ : SUCCESS in 16m 39s

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1964

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1712/ : SUCCESS in 4m 33s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1993/ : SUCCESS in 54s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/790/ : SUCCESS in 38m 43s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7727/ : SUCCESS in 20m 13s

**Pull request merged**